specs/derivation.md

```diff
@@ -27,181 +43,606 @@ TODO
-Others can then reproduce the same L2 changes.
+However, in the very worst case, we can only reconstruct the batch for the first L2 block in the epoch by reading the
+last L1 block of the sequencing window. This means we also can't derive any further L2 block in the epoch until then, as
+they need the state that results from applying the epoch's first L2 block.
 
-Consuming the stream presents an inductive process:
-given that we know the last L2 block derived from the previous [sequencing window][g-sequencing-window], as well as the
-next [sequencing window][g-sequencing-window], then we can derive [payload attributes] of the next L2 blocks.
+------------------------------------------------------------------------------------------------------------------------
 
-To derive the whole L2 chain from scratch, we simply start with the L2 genesis block as the last L2 block, and the
-block at height `L2_CHAIN_INCEPTION + 1` as the start of the next sequencing window.
-Then we iteratively apply the derivation process, using the [derivation pipeline](#derivation-pipeline) process outlined below.
+# Batch Submission
 
-> **TODO** specify genesis block
+## Sequencing & Batch Submission Overview
 
-## Rollup Stream format
+The [sequencer][g-sequencer] accepts L2 transactions from users. It is responsible for building blocks out of these. For
+each such block, it also creates a corresponding [sequencer batch][g-sequencer-batch]. It is also responsible for
+submitting each batch to a [data availability provider][g-avail-provider] (e.g. Ethereum calldata), which it does via
+its [batcher][g-batcher] component.
 
-The stream is ***multiplexed*** and ***buffered***, to handle serve the required properties:
+The difference between an L2 block and a batch is subtle but important: the block includes an L2 state root, whereas the
+batch only commits to transactions at a given L2 timestamp (equivalently: L2 block number).
 
-- There may be different concurrent data submissions to L1
-- There may be different actors that submit the data, the system cannot rely on a single EOA nonce value.
-- The fraud proof processes submitted data, which has to be bounded
-- The submissions confirmed on L1 may be confirmed out of order
-- The L1 may [reorganize][g-reorg], requiring the submissions to reset to a previous point
-- The L2 grows, and does so in smaller intervals than L1
-- The L2 may have bursts that cannot be fully confirmed in a single piece of submitted data
+This means that even if the sequencer applies a state transition incorrectly, the transactions in the batch will stil be
+considered part of the canonical L2 chain. Batches are still subject to validity checks (i.e. they have to be encoded
+correctly), and so are individual transactions within the batch (e.g. signatures have to be valid). Invalid batches and
+invalid individual transactions within an otherwise valid batch are discarded by correct nodes.
```

Comment:
TODO: check whether the whole gets dropped if we have a bad transaction
I reckon we should only drop the invalid transaction.

Reply:
Only invalid transactions get dropped.